spec/models/course_user_spec.rb

```diff
@@ -155,14 +156,25 @@
     end
 
     describe '#staff?' do
-      it 'returns true if the role is teaching assistant, manager or owner' do
+      it 'returns true if the role is observer, teaching assistant, manager or owner' do
         expect(student.staff?).to be_falsey
+        expect(observer.staff?).to be_truthy
         expect(teaching_assistant.staff?).to be_truthy
         expect(manager.staff?).to be_truthy
         expect(course_owner.staff?).to be_truthy
       end
     end
 
+    describe '#teaching_staff?' do
+      it 'returns true if the role is teaching assistant, manager or owner' do
+        expect(student.teaching_staff?).to be_falsey
+        expect(observer.teaching_staff?).to be_falsey
+        expect(teaching_assistant.teaching_staff?).to be_truthy
+        expect(manager.teaching_staff?).to be_truthy
+        expect(course_owner.teaching_staff?).to be_truthy
+      end
+    end
+
     describe '#manager_or_owner?' do
```

Comment:
Not that important, but want to also add tests for observers for `#manager_or_owner?` and `#real_student?` for the sake of completeness?